util/llpcElfWriter.cpp

```diff
@@ -955,6 +966,118 @@ void ElfWriter<Elf>::MergeElfBinary(
     WriteToBuffer(pPipelineElf);
 }
 
+template<class Elf>
+void ElfWriter<Elf>::Reinitialize() {
+    m_map.clear();
+    m_notes.clear();
+    m_symbols.clear();
+
+    m_sections.push_back({});
+
+    m_strtabSecIdx = m_sections.size();
+    m_sections.push_back({});
+    m_map[StrTabName] = m_strtabSecIdx;
+
+    m_textSecIdx = m_sections.size();
+    m_sections.push_back({});
+    m_map[TextName] = m_textSecIdx;
+
+    m_symSecIdx = m_sections.size();
+    m_sections.push_back({});
+    m_map[SymTabName] = m_symSecIdx;
+
+    m_noteSecIdx = m_sections.size();
+    m_sections.push_back({});
+    m_map[NoteName] = m_noteSecIdx;
+}
+
+// TODO: Replace Elf64 with a template.  I did this for now to make the autocomplete in my ide work better.
```

Comment:
I guess this is already done?

Aside: I find the use of templates here highly questionable. But it's not your doing, so it's fine :)

Also, I think &&-refs on relocatableElf need some justification. Why not just pass an ArrayRef?